### PR TITLE
Change specific CArray type to std::array (c++11)

### PR DESCRIPTION
**Pull request by untereiner**
I did this change on my own clone.
It could maybe interest you.

### CODE REVIEW

File: include/nanoflann.hpp [hunk 1]
```diff
@@ -70,6 +71,38 @@ namespace nanoflann
...
   *  @{ */
 
+	/**
+	 * Traits if object is resizable (typically has a resize method)
+	 */
+  	template <typename T, typename = int>
+	struct is_resizable : std::false_type {};
+
+	template <typename T>
+	struct is_resizable <T, decltype((void) &T::resize, 0)> : std::true_type {};
```
Comment (jlblancoc) [inline include/nanoflann.hpp]: Loved this one!

**Comment by jlblancoc** [comment]
:+1: to the idea, thank you very much! It totally makes sense now that we raised the requisite to C++11. 

However, there is a failing test [here](https://travis-ci.org/jlblancoc/nanoflann/jobs/353859671#L590), could you take a look at it?

**Comment by svenevs** [comment]
There's no job log? Sometimes Travis gets confused. You could try closing and reopening the pull request to relaunch Travis if no job log is showing for you as well.

**Comment by untereiner** [comment]
I have a wired core dump when I try to correct the code. I am looking at it.

**Comment by untereiner** [comment]
Everything is fine now... at least on linux

**Comment by jlblancoc** [comment]
:+1: Nice contribution. Thanks again!

**Pull request merged**